### PR TITLE
JavaScript Tools Every Developer Should Know

### LINKED ISSUE #6008: [Languages] JavaScript Tools Every Developer Should Know
### Topic Suggestion
[Languages] JavaScript Tools Every Developer Should Know

#### Proposed title of article
JavaScript Tools Every Developer Should Know
#### Proposed article introduction
JavaScript is still the most widely used programming language on the planet. You'll almost certainly require JavaScript if you want to develop anything on the web. The language's success is partly due to the numerous JavaScript tools that make programming simple and pleasant. These tools are intended to make development easier and to aid with debugging. The JavaScript tool you require depends on the task you're attempting to do.

There are plenty of JavaScript tools available. The challenge is s deciding which ones are best for you and your workflow. This article introduces some of the best JavaScript development tools, frameworks, and libraries to help developers be more productive.

#### Key takeaways 
1. Web Components
2. Gatsby
3. Gulp
4. ECMAScript modules (ESM)
5. WebPack
6. NPM
7. ESLint
8. Vite
9. Frameworks
10. Libraries

#### Article quality
This article will dig deeper into JavaScript Development tools. JavaScript has thousands of useful tools associated with it. This guide will share 10+ tools that JavaScript developers at all levels can use to improve their productivity.

#### References
Please list links to any published content/research that you intend to use to support/guide this article.

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by LeahKui**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #6008

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

File: content/articles/javascript-tools-every-developer-should-know/index.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
+JavaScript is still the most widely used programming language on the planet. You'll almost certainly require JavaScript if you want to develop anything on the web. The language's success is partly due to the numerous JavaScript tools that make programming simple and pleasant. JavaScript tools are designed to make development and debugging easier. The JavaScript tool you require depends on the task you're attempting to do.
```
Comment (hectorkambow) [inline content/articles/javascript-tools-every-developer-should-know/index.md:1]: @LeahKui a  statement like this needs a source

**Comment by LeahKui** [comment]
@hectorkambow I added a source

**Comment by ahmadmardeni1** [comment]
Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
Be sure to see our resources page to see more info on plagiarism and what is considered as such.

[JavaScript Tools Every Developer Should Know #6689](https://github.com/section-engineering-education/engineering-education/files/8151649/New.Scan.8_12.AM.Copyleaks.Report.pdf)

**Comment by LeahKui** [comment]
@ahmadmardeni1 , @BrianaNzivu I worked on plagiarism errors

**Comment by ahmadmardeni1** [comment]
Plagiarism check done ✅

**Review by BrianaNzivu — APPROVED** [review]
Hey @LeahKui   👋 .

Very good job on the article. It is a very interesting read. Barely any changes to be made 👏 . 
- Improve your sentence structure and arrangement. It is causing clarity issues in your article.
- Please go through ALL the corrections I have made so that you can not repeat the same mistakes.

That's it! :smile: 

A final review will be done now by a content moderator.
Otherwise, congratulations on publishing your article🍾.

In case of any question or clarification (currently or in the future), do not hesitate to reach out via [Slack](https://app.slack.com/client/T5X14SH7G/D01A6MT673R) or my personal email (briananzivu@gmail.com). 
I am **ALWAYS** here to help.

It has been a pleasure reading and reviewing this article.

Looking forward to seeing you contribute more to EngEd. 🚀

Kind Regards.

**Comment by hectorkambow** [comment]
jumping on this one 👍

**Pull request merged**